### PR TITLE
Eliminate properties used for publishing artifacts

### PR DESCRIPTION
**Pull request by JakeWharton**
The project name is now used as the `artifactId` automatically. These all matched the explicitly-specified ones from before.

I have eliminated the name which is now also set to the project name. Prior to this it was set to a more human-readable version of the `artifactId`.

This was tested by installing all artifacts to a local Maven repo both before and after this change and then diffing the directories. The only changes to the poms is the name change. For example:

    diff -r build/localMavenOld/app/cash/redwood/redwood-treehouse-js/0.5.0/redwood-treehouse-js-0.5.0.pom build/localMaven/app/cash/redwood/redwood-treehouse-js/0.5.0/redwood-treehouse-js-0.5.0.pom
    13c13
    <   <name>Redwood Treehouse</name>
    ---
    >   <name>redwood-treehouse</name>
    15a16

### CODE REVIEW

**Review by colinrtwhite — APPROVED** [review]
Nice! We can also delete the `gradle.properties` in the new `redwood-yoga` module I just added.

**Comment by JakeWharton** [comment]
Yep. Done!

File: build-support/src/main/kotlin/app/cash/redwood/buildsupport/RedwoodBuildPlugin.kt [hunk 1]
```diff
@@ -96,6 +103,37 @@ private class RedwoodBuildExtensionImpl(private val project: Project) : RedwoodB
       if (project.providers.systemProperty("RELEASE_SIGNING_ENABLED").getOrElse("true").toBoolean()) {
         signAllPublications()
       }
+
+      coordinates(redwoodGroupId, project.name, redwoodVersion)
+
+      pom { pom ->
```
Comment (swankjesse) [inline build-support/src/main/kotlin/app/cash/redwood/buildsupport/RedwoodBuildPlugin.kt:109]: YAY

**Review by swankjesse — APPROVED** [review]
LOVE THIS MUCHLY

**Comment by veyndan** [comment]
🥳

**Pull request merged**